src/core/evaluator.js

```diff
@@ -612,16 +615,75 @@ class PartialEvaluator {
       );
       const decode = dict.getArray("D", "Decode");
 
+      if (this.parsingType3Font) {
+        imgData = PDFImage.createMaskForType3({
+          imgArray,
+          width: w,
+          height: h,
+          imageIsFromDecodeStream: image instanceof DecodeStream,
+          inverseDecode: !!decode && decode[0] > 0,
+          interpolate,
+        });
+
+        imgData.cached = !!cacheKey;
+        args = [imgData];
+
+        operatorList.addOp(OPS.paintImageMaskXObject, args);
+        if (cacheKey) {
+          localImageCache.set(cacheKey, imageRef, {
+            fn: OPS.paintImageMaskXObject,
+            args,
+          });
+        }
+
+        if (optionalContent !== undefined) {
+          operatorList.addOp(OPS.endMarkedContent, []);
+        }
+        return;
+      }
+
       imgData = PDFImage.createMask({
         imgArray,
         width: w,
         height: h,
-        imageIsFromDecodeStream: image instanceof DecodeStream,
         inverseDecode: !!decode && decode[0] > 0,
         interpolate,
+        imageIsFromDecodeStream: image instanceof DecodeStream,
       });
+
+      if (imgData.isSingleOpaquePixel) {
+        // Handles special case of mainly LaTeX documents which use image
+        // masks to draw lines with the current fill style.
+        operatorList.addOp(OPS.paintSolidColorImageMask, []);
+        if (cacheKey) {
+          localImageCache.set(cacheKey, imageRef, {
+            fn: OPS.paintSolidColorImageMask,
+            args: [],
+          });
+        }
+
+        if (optionalContent !== undefined) {
+          operatorList.addOp(OPS.endMarkedContent, []);
+        }
+        return;
+      }
+
       imgData.cached = !!cacheKey;
-      args = [imgData];
+
+      const objId = `mask_${this.idFactory.createObjId()}`;
+
+      if (imgData.bitmapPromise) {
+        const { bitmapPromise } = imgData;
+        delete imgData.bitmapPromise;
+        bitmapPromise.then(bitmap => {
+          imgData.bitmap = bitmap;
+          this._sendImgData(objId, imgData);
+        });
+      } else {
+        this._sendImgData(objId, imgData);
+      }
+      operatorList.addDependency(objId);
+      args = [{ data: objId, width: w, height: h }];
```

Comment:
You're not passing along the `interpolate` parameter here.

While it might not be necessary for this particular data, it still seems nice to be consistent w.r.t. the API. Also, I'd probably suggest actually returning the parameters from the `PDFImage.createMask`-call itself to avoid any future "surprises" here:
```javascript
args = [{
  data: objId,
  width: imgData.width,
  height: imgData.height,
  interpolate: imgData.interpolate,
}];
```